ffi/framebuffer_android.lua

```diff
@@ -70,6 +70,33 @@ function framebuffer:resize()
     self:_updateWindow()
 end
 
+function framebuffer:getRotationMode()
+    if android.hasNativeRotation() then
+        return android.orientation.get()
+    else
+        return self.cur_rotation_mode
+    end
+end
+
+function framebuffer:setRotationMode(mode)
+    if android.hasNativeRotation() then
+        local key
+        if mode == 0 then key = "PORTRAIT"
+        elseif mode == 1 then key = "LANDSCAPE"
+        elseif mode == 2 then key = "REVERSE_PORTRAIT"
+        elseif mode == 3 then key = "REVERSE_LANDSCAPE" end
+        if key then
+            android.orientation.set(C["ASCREEN_ORIENTATION_" .. key])
+        end
+    else
+        self.bb:rotateAbsolute(-90 * (mode - self.native_rotation_mode - self.blitbuffer_rotation_mode))
```

Comment:
Heh, what a line. :-)

Reply:
That's the generic `fb:setRotationMode` too ^^

Reply:
Yup. Not sure how to call the parent function from a child implementation in lua. But yeah if android.isNativeRotation() is false we use the parent `setRotationMode` function.

Reply:
@pazos There's no default mechanism, but note what we do in:

https://github.com/koreader/koreader-base/blob/68b3a7c5d74cb77d82a9eff5b9d9dd9c5bb368f7/ffi/framebuffer.lua#L59-L65

As such self.parent should do the trick.

Reply:
I just realized it's widely used. ;-)
https://github.com/koreader/koreader-base/blob/68b3a7c5d74cb77d82a9eff5b9d9dd9c5bb368f7/ffi/framebuffer_android.lua#L59

Reply:
Tried with

```diff
diff --git a/ffi/framebuffer_android.lua b/ffi/framebuffer_android.lua
index c22fc87..72f6154 100644
--- a/ffi/framebuffer_android.lua
+++ b/ffi/framebuffer_android.lua
@@ -89,11 +89,14 @@ function framebuffer:setRotationMode(mode)
             android.orientation.set(C["ASCREEN_ORIENTATION_" .. key])
         end
     else
+        self.parent:setRotationMode(mode)
+--[[
         self.bb:rotateAbsolute(-90 * (mode - self.native_rotation_mode - self.blitbuffer_rotation_mode))
         if self.viewport then
             self.full_bb:setRotation(self.bb:getRotation())
         end
         self.cur_rotation_mode = mode
+--]]
     end
 end
```

Result:
```
07-05 12:19:44.229  4830  4845 E KOReader: [NativeThread]: failed to run script: ffi/framebuffer.lua:312: attempt to index field 'bb' (a nil value)
07-05 12:19:44.229  4830  4845 E KOReader: [NativeThread]: Stopping due to previous errors
```

Reply:
What about `framebuffer.parent.setRotationMode(self, mode)`

Reply:
Tried that too. Even funnier:

```
07-05 12:17:49.810  4704  4719 E KOReader: [NativeThread]: failed to run script: stack overflow
07-05 12:17:49.810  4704  4719 E KOReader: [NativeThread]: Stopping due to previous errors
```

Reply:
@NiLuJe's suggestion should work. What did you try exactly?

Reply:
Indeed. I tried self.parent.setRotationMode(self, mode) because I have no idea about inheritance :facepalm: